core/plannernotes.c

```diff
@@ -30,16 +30,65 @@ int diveplan_duration(struct diveplan *diveplan)
 	return (duration + 30) / 60;
 }
 
+struct icd_data { // This structure provides communication between function isobaric_counterdiffusion() and the calling software.
+	int time; 			// For a trimix gas change, the time (in seconds) of the change
+	int ambientpressure_mbar;	// The ambient pressure (mbar) when the gas change took place
+	int dN2;			// The change in fraction (permille) of nitrogen during the change
+	int dHe;			// The change in fraction (permille) of helium during the change
+};
+
+/* Perform isobaric counterdiffusion calculations for gas changes in trimix dives.
+ * Here we use the rule-of-fifths where, during a change involving trimix gas, the increase in nitrogen
+ * should not exceed one fifth of the decrease in helium.
+ * Parameters:	1) Pointer to the dive structure.
+ *				2) Pointer to the dive planner point where the gas change is taking place.
+ *				3) pointers to two gas mixes, the gas being switched from and the gas being switched to.
+ * 				4) a pointer to an icd_data structure.
+ * Output:		 i) The icd_data stucture is filled with the time, pressure, delta_N2 and delta_He numbers (as permille).
+ * 				ii) Function returns a boolean indicating an exceeding of the rule-of-fifths. False = no problem.
+ */
+bool isobaric_counterdiffusion(struct dive *dive, struct divedatapoint *dp, struct gasmix *oldgasmix, struct gasmix *newgasmix, struct icd_data *results)
```

Comment:
Please don't make dp an argument: This only exists for planned dives (and we could do this test also for gas changes in real logged dives) and you only use it to copy the time to the icd_data struct (where I honestly don't really understand why you need the time there anyway, you could pass that as an extra argument to add_icd_entry).

Reply:
done